File: solutions/0414-Third-Maximum-Number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const thirdMax = function(nums) {
  const set = new Set(nums.sort((a,b) => a - b));
  
  return (set.size < 3) ? [...set].pop() : [...set].reverse()[2];
};

File: solutions/0414-Third-Maximum-Number.cpp
class Solution {
public:
  int thirdMax(vector<int>& nums) {
    sort(nums.begin(), nums.end(), greater<int>());
    set<int, greater<int>> s(nums.begin(), nums.end());
    
    auto it = s.begin();
    advance(it, 2);
    
    return s.size() < 3 ? *(s.begin()) : *it;
  }
};
